### ISSUE TITLE
[BUG] "Failed to fetch", but no explanation

### ISSUE THREAD

**Issue by celeryclub** [microsoft/playwright#1397]
**Context:**
- Playwright Version: 0.11.1-next.1583909126688
- Operating System: macOS

**Code Snippet**

```javascript
const { chromium, webkit, firefox } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: false });
  const context = await browser.newContext({ bypassCSP: true });
  const page = await context.newPage();

  await page.goto('https://www.behance.net', { waitUntil: 'networkidle0' });

  const myText = await page.evaluate(async () => {
    const response = await fetch('https://example.com');
    const text = await response.text();
    return text;
  });

  console.log(myText);

  await browser.close();
})();
```

**Describe the bug**
Hi team. I'm seeing this error when I try to run the above code.
`UnhandledPromiseRejectionWarning: Error: Evaluation failed: TypeError: Failed to fetch`

The request I'm making is cross-origin, so this could be a CSP or CORS error, but the error message doesn't give me any information about what the error is. It's possible this is out of scope for what Playwright is doing, but I thought I'd at least check. Thank you!

**Comment by pavelfeldman** [comment]
Sadly, `Failed to fetch` is the only information available in the exception itself. The rest is available in Chrome via the console. You can too log what is getting into the console:

`page.on('console', m => console.log(m.text()));`

That way you get this:

```
Access to fetch at 'https://example.com/' from origin 'https://www.behance.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
Failed to load resource: net::ERR_FAILED
```

I wonder if we should provide an option that translates console messages / errors into the node console with less effort.

**Comment by celeryclub** [comment]
Thank you for the explanation @pavelfeldman. That does sound like it could be a nice feature. I'll go ahead and close this since it seems like a separate thing.

**Comment by dustinvo17** [comment]
Hi @celeryclub, were you able to figure the error? I have successfully run the snippet of code but when I tried to run it on a pipeline it ran into same issue -> page.evaluate: Evaluation failed: TypeError: Failed to fetchError:
- Here is my piece of code 
await page.evaluate(async (apiURL) => {
            const file = new File(["backup"], "userBackup.json", {
                type: "application/json",
            });

            const formData = new FormData();
            formData.append("userBackup", file);

            const createBackupResponse = await fetch(`${apiURL}/api/v1/backups`, {
                method: "POST",
                body: formData,
            });

            return await createBackupResponse.json();
        }, this.apiURL);
